insanely_fast_whisper_rocm/core/integrations/diarization.py

```
@@ -0,0 +1,449 @@
+
+    Args:
+        exc: Runtime exception raised by pyannote/PyTorch.
+
+    Returns:
+        True when the error matches a known ROCm/MIOpen runtime failure.
+    """
+    message = str(exc).lower()
+    return (
+        "miopenstatusunknownerror" in message
+        or "miopen" in message
+        and "rocrand" in message
```

Ruff (0.15.12)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)